tests/integration/common/data/db/test_constraints.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from app.common.data.models import RoleEnum
+
+
+class TestUserRoleConstraints:
+    def test_member_role_not_platform(self, db_session, factories):
+        user_role = factories.user_role.create(has_grant=False, has_organisation=False, role=RoleEnum.MEMBER)
+        db_session.add(user_role)
+        with pytest.raises(IntegrityError) as error:
+            db_session.flush()
+        assert (
+            'new row for relation "user_role" violates check constraint "ck_user_role_member_role_not_platform"'
+            in error.value.args[0]
+        )
+
+    @pytest.mark.parametrize("has_organisation, has_grant", [(True, True), (False, True)])
+    def test_s151_officer_role_org_only(self, db_session, factories, has_organisation, has_grant):
+        user_role = factories.user_role.create(
+            has_organisation=has_organisation, has_grant=has_grant, role=RoleEnum.S151_OFFICER
+        )
+        db_session.add(user_role)
+        with pytest.raises(IntegrityError) as error:
+            db_session.flush()
+        assert (
+            'new row for relation "user_role" violates check constraint "ck_user_role_s151_officer_role_org_only"'
+            in error.value.args[0]
+        )
+
+    @pytest.mark.parametrize("has_organisation, has_grant", [(True, True), (True, False)])
+    def test_assessor_role_grant_only(self, db_session, factories, has_organisation, has_grant):
+        user_role = factories.user_role.create(
+            has_organisation=has_organisation, has_grant=has_grant, role=RoleEnum.ASSESSOR
+        )
+        db_session.add(user_role)
+        with pytest.raises(IntegrityError) as error:
+            db_session.flush()
```

Comment:
hmm something about this feels off - I thought that `.create()` on a factory should already add and flush/commit it to the session 🤔